GVRf/Framework/jni/engine/importer/assimp_importer.cpp

```diff
@@ -68,4 +71,176 @@ std::shared_ptr<Mesh> AssimpImporter::getMesh(int index) {
+                    material->setTexture("main_texture", texture);
+                    // Sets the default main texture
+
+                } else {
+                    if ((ai_material->GetTextureCount(aiTextureType_DIFFUSE)) <= 0) {
+                        // No texture file found so apply default texture
+                        std::shared_ptr<Texture> texture(new BaseTexture(env, bitmap));
+                        material->setTexture("main_texture", texture);
+                    } else {
+                        // About to enter Java side.
+                        jobject mBitmap = env->CallStaticObjectMethod(texClass, mID, texFileName);
+                        if(mBitmap == NULL)
+                        {
+                            // Null bitmap for texture file for mesh
+                            // Applying default texture
+                            std::shared_ptr<Texture> texture(new BaseTexture(env, bitmap));
+                            material->setTexture("main_texture", texture);
+                        }
+                        else
+                        {
+                            // Back to native side from Java side
+                            // Applying actual texture
+                            std::shared_ptr<Texture> texture(new BaseTexture(env, mBitmap));
+                            material->setTexture("main_texture", texture);
+                        }
+                    }
+                }
+            }
+
+            // Set the material to the render data
+            renderData->set_material(material);
+
+            // Transformation
+            std::shared_ptr<Transform> transform(new Transform());
+            transform->set_owner_object(scnObject);
+
+            // Accumulated transformations of the node
+            aiMatrix4x4 aTransform = node->mTransformation * accumulatedTransform;
+            aiVector3t<float> scaling;
+            aiQuaterniont<float> rotation;
+            aiVector3t<float> position;
+            aTransform.Decompose(scaling, rotation, position);
+            int meshIndex = i;
+            transform->set_position(position.x, position.y, position.z);
+            transform->set_rotation(rotation.w, rotation.x, rotation.y, rotation.z);
+            transform->set_scale(scaling.x, scaling.y, scaling.z);
+            scnObject->attachTransform(scnObject, transform);
+            scnObject->attachRenderData(scnObject, renderData);
+
+            //Attaches the Scene Object to the Scene.
+            scnPtr->addSceneObject(scnObject);
+        } else {
+            //new RenderData class and perform all the operations
+        }
+    }
+
+    for(int i=0;i<node->mNumChildren;i++)
+    {
+        sceneRecursion(node->mChildren[i], aScene, scnPtr, env, obj, bitmap, accumulatedTransform);
+    }
+}
+
+
+std::shared_ptr<Scene> AssimpImporter::loadScene(JNIEnv* env, jobject obj, jobject bitmap)
+{
+    //  Scene
+    std::shared_ptr<Scene> scenePtr(new Scene());
+
```

Comment:
This was my major concern from your previous patch.
I am already seeing these code re-uses in the new patch, including this:
+ float camera_separation_distance_ = camera_rig-default_camera_separation_distance(); 

May be we can further do code modularizartions in the future as the features grow. Thanks.